dev/lint-python

```diff
@@ -122,6 +123,32 @@ function pycodestyle_test {
     fi
 }
 
+function mypy_test {
+    local MYPY_REPORT=
+    local MYPY_STATUS=
+
+    if ! hash "$MYPY_BUILD" 2> /dev/null; then
+        echo "The mypy command was not found."
+        echo "mypy checks failed."
+        # exit 1
+        pip3 install mypy
```

Comment:
Oh let's don't do this.

Reply:
We should open a separate ticket to do this in Jenkins and ask @shaneknapp to install

Reply:
I've created a ticket in Jenkins: https://issues.apache.org/jira/browse/SPARK-32797